core/src/main/scala/co/upvest/arweave4s/adt/Address.scala

```diff
@@ -2,10 +2,17 @@ package co.upvest.arweave4s.adt
 
 import java.security.interfaces.RSAKey
 import co.upvest.arweave4s.utils.{CryptoUtils, UnsignedBigInt}
+import scala.util.Try
 
-class Address(val bytes: Array[Byte]) extends Base64EncodedBytes
+class Address protected (val bytes: Array[Byte]) extends Base64EncodedBytes {
+  require(bytes.length == Address.Length)
```

Comment:
Proposal:
Remove line 8.
Change line 14 to: `Try{require(bytes.length == Address.Length) map (new Address(_))}`
Adress constructor can be made `private` as well. I would consider this a cleaner solution.